pkg/apis/migration/v1alpha1/migplan_types.go

```diff
@@ -416,32 +347,57 @@ func (r *MigPlan) UpdateRegistryDC(storage *MigStorage, deploymentconfig *appsv1
 		}
 		envFrom = append(envFrom, source)
 	}
-	deploymentconfig.Spec = appsv1.DeploymentConfigSpec{
-		Replicas: 1,
-		Selector: map[string]string{
-			"app":              name,
-			"deploymentconfig": name,
-		},
-		Strategy: appsv1.DeploymentStrategy{Resources: kapi.ResourceRequirements{}},
-		Template: &kapi.PodTemplateSpec{
+
+	deployment.Spec = appsv1.DeploymentSpec{
+		Replicas: pointer.Int32Ptr(1),
+		Selector: metav1.SetAsLabelSelector(map[string]string{
+			"app":        name,
+			"deployment": name,
+			"migplan":    r.Name,
+		}),
+		Template: kapi.PodTemplateSpec{
 			ObjectMeta: metav1.ObjectMeta{
 				CreationTimestamp: metav1.Time{},
 				Labels: map[string]string{
-					"app":              name,
-					"deploymentconfig": name,
+					"app":        name,
+					"deployment": name,
+					"migplan":    r.Name,
 				},
 			},
 			Spec: kapi.PodSpec{
 				Containers: []kapi.Container{
 					kapi.Container{
-						EnvFrom: envFrom,
-						Image:   registryImage,
-						Name:    "registry",
+						EnvFrom:                  envFrom,
+						Image:                    registryImage,
+						Name:                     "registry",
+						TerminationMessagePolicy: kapi.TerminationMessageFallbackToLogsOnError,
 						Ports: []kapi.ContainerPort{
 							kapi.ContainerPort{
 								ContainerPort: 5000,
 								Protocol:      kapi.ProtocolTCP,
 							},
+							kapi.ContainerPort{
+								ContainerPort: 5001,
+								Protocol:      kapi.ProtocolTCP,
+							},
+						},
+						LivenessProbe: &kapi.Probe{
```

Comment:
Looks like we decided to go with a liveness and a readiness probe here. Is it expected that the registry container will always be `live` on the `/v2/_catalog` path in 5 seconds, or do you think it makes sense to use `initialDelaySeconds` to provide a grace period on initial boot?